src/ddsrt/include/dds/ddsrt/process.h

```diff
@@ -41,6 +44,98 @@ typedef pid_t ddsrt_pid_t;
 DDS_EXPORT ddsrt_pid_t
 ddsrt_getpid(void);
 
+#ifdef DDSRT_USE_PROCESSCREATION
+#define DDSRT_HAVE_PROCESSCREATION 1
+
+/**
+ * @brief Create new process.
+ *
+ * Creates a new process using the provided executable file. It will have
+ * default priority and scheduling.
+ *
+ * Process arguments are represented by argv, which can be null. If argv is
+ * not null, then the array must be null terminated. The argv array only has
+ * to contain the arguments, the executable filename doesn't have to be in
+ * the first element, which is normally the convention.
+ *
+ * @param[in]   executable     Executable file name.
+ * @param[in]   argv           Arguments array.
+ * @param[out]  pid            ID of the created process.
+ *
+ * @returns A dds_retcode_t indicating success or failure.
+ *
+ * @retval DDS_RETCODE_OK
+ *             Process successfully created.
+ * @retval DDS_RETCODE_BAD_PARAMETER
+ *             Provided file is not executable.
+ * @retval DDS_RETCODE_ERROR
+ *             Process could not be created.
+ */
+DDS_EXPORT dds_retcode_t
+ddsrt_process_create(
+  const char *executable,
+  char *const argv[],
+  ddsrt_pid_t *pid);
+
+/**
+ * @brief Wait for a process to exit.
+ *
+ * When the process (identified by pid) has exited, the status argument will
+ * contain the process exit code.
+ *
+ * When the process is terminated by a call to ddsrt_process_terminate(), then
+ * the return code is uncertain, except that it won't be DDS_RETCODE_TIMEOUT.
+ *
+ * @param[in]   pid            Process ID (PID) of the process to wait for.
+ * @param[in]   timeout        Max timeout to wait for the process to exit.
+ * @param[out]  status         The exit code of the process.
+ *
+ * @returns A dds_retcode_t indicating success or failure.
+ *
+ * @retval DDS_RETCODE_OK
+ *             Process has exited.
+ * @retval DDS_RETCODE_TIMEOUT
+ *             Process is still alive.
+ * @retval DDS_RETCODE_BAD_PARAMETER
+ *             Process unknown.
+ * @retval DDS_RETCODE_ERROR
+ *             Wait failed for an unknown reason.
+ */
+DDS_EXPORT dds_retcode_t
+ddsrt_process_wait_exit(
```

Comment:
I'd prefer this to be renamed to `ddsrt_waitpid` and that specifying DDS_INFINITY for timeout, which I'd prefer to be named `reltime`, waits indefinitely for the process to exit.

Reply:
This has been refactored. The way it's (very likely) going to be used in the multi test framwork is closer to the Windows GetExitCodeProcess() than the Posix waitpid: only check a single process without ever using a timeout (certainly not 'infinite').
So, ddsrt_process_wait_exit() has been changed into ddsrt_process_get_exit_code().